test_runner/regress/test_ondemand_slru_download.py

```diff
@@ -0,0 +1,113 @@
+from fixtures.log_helper import log
+from fixtures.neon_fixtures import NeonEnvBuilder
+from fixtures.types import Lsn
+from fixtures.utils import query_scalar
+
+
+#
+# Test on-demand download of the pg_xact SLRUs
+#
+def test_ondemand_download_pg_xact(neon_env_builder: NeonEnvBuilder):
```

Comment:
Could you parametrize this test to have a sharded variant? 

Right now we store SLRUs on all shards, but that will probably change & I'd like to make sure we are covered against regressions by ensuring that compute is always requesting SLRUs from shard 0 (I don't know if it actually does that currently)

Reply:
Ok, added a `shard_count` parameter to the tests, copying from how it's done in `test_pg_regress.py`.

> Right now we store SLRUs on all shards, but that will probably change & I'd like to make sure we are covered against regressions by ensuring that compute is always requesting SLRUs from shard 0 (I don't know if it actually does that currently)

Yes, the SLRU download requests are always sent to shard 0 (https://github.com/neondatabase/neon/blob/139d1346d5aed41e1cf1479343943f9bf3670794/pgxn/neon/pagestore_smgr.c#L2837)